modules/cert-manager-prepare-cloud-credentials-aws-sts.adoc

```diff
@@ -0,0 +1,146 @@
+    apiVersion: cloudcredential.openshift.io/v1
+    kind: AWSProviderSpec
+    statementEntries:
+    - action:
+      - "route53:GetChange"
+      effect: Allow
+      resource: "arn:aws:route53:::change/*"
+    - action:
+      - "route53:ChangeResourceRecordSets",
+      - "route53:ListResourceRecordSets"
+      effect: Allow
+      resource: "arn:aws:route53:::hostedzone/*"
+    - action:
+      - "route53:ListHostedZonesByName"
+      effect: Allow
+      resource: "*"
+  secretRef:
+    name: aws-creds
+    namespace: cert-manager
+  serviceAccountNames:
+  - cert-manager
+----
+
+. Create a `CredentialsRequest` resource by running the following command:
++
+[source,terminal]
+----
+$ oc create -f sample-credential-request.yaml
+----
+
+. Use the `ccoctl` tool to process `CredentialsRequest` objects of the {cert-manager-operator} by running the following command:
++
+[source,terminal]
+----
+$ ccoctl aws create-iam-roles \
+    --name <name> --region=<aws_region> \
+    --credentials-requests-dir=<path-to-credrequests-dir> \
+    --identity-provider-arn <oidc-arn>
+----
++
+.Example output
+[source,terminal]
+----
+2023/05/15 18:10:34 Role arn:aws:iam::XXXXXXXXXXXX:role/<oidc-prefix>-cert-manager-aws-creds created
+2023/05/15 18:10:34 Saved credentials configuration to: <output-dir>/manifests/cert-manager-aws-creds-credentials.yaml
+2023/05/15 18:10:35 Updated Role policy for Role <oidc-prefix>-cert-manager-aws-creds
+----
++
+Copy the `<aws-arn-id>` from the output to use in the next step.
+
+. Add the `eks.amazonaws.com/role-arn="<aws-arn-id>"` annotation to the service account by running the following command:
++
+[source,terminal]
+----
+$ oc -n cert-manager annotate serviceaccount cert-manager eks.amazonaws.com/role-arn="<aws-arn-id>"
+----
+
+. Add the `eks.amazonaws.com/audience="sts.amazonaws.com"` annotation to the service account by running the following command:
++
+[source,terminal]
+----
+$ oc -n cert-manager annotate serviceaccount cert-manager eks.amazonaws.com/audience="sts.amazonaws.com"
+----
+
+. Add the `eks.amazonaws.com/sts-regional-endpoints="true"` annotation to the service account by running the following command:
++
+[source,terminal]
+----
+$ oc -n cert-manager annotate serviceaccount cert-manager eks.amazonaws.com/sts-regional-endpoints="true"
+----
+
+. Add the `eks.amazonaws.com/token-expiration="86400"` annotation to the service account by running the following command:
++
+[source,terminal]
+----
+$ oc -n cert-manager annotate serviceaccount cert-manager eks.amazonaws.com/token-expiration="86400"
+----
+
+. Apply the secrets generated in the manifests directory of your cluster by running the following command:
++
+[source,terminal]
+----
+$ ls manifests/*-credentials.yaml | xargs -I{} oc apply -f {}
+----
+
+. Verify that the credentials secret of the {cert-manager-operator} is created by running the following command:
++
+[source,terminal]
+----
+$ oc get deployment cert-manager -n cert-manager
```

Comment:
Please tweak this one to be instead, as AWS specific env vars will only be on the pod and not on the deployment:
```
$ oc get pods -n cert-manager -l app.kubernetes.io/name=cert-manager | grep AWS
```